src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -163,10 +163,11 @@ public class ControllerArgumentBinder : IControllerArgumentBinder
                 parameter.BindingInfo,
                 parameter.Name);
 
-            if (parameter.BindingInfo?.BinderModelName != null)
+            var parameterModelName = parameter.BindingInfo?.BinderModelName ?? metadata.BinderModelName;
```

Comment:
I was a little surprised that this was a thing, but apparently it is supported by all the infrastructure...